plugins/permission/src/components/PermissionedRoute.test.tsx

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { PermissionedRoute } from '.';
+import { usePermission } from '../hooks';
+import { renderInTestApp } from '@backstage/test-utils';
+
+jest.mock('../hooks', () => ({
+  usePermission: jest.fn(),
+}));
+const mockUsePermission = usePermission as jest.MockedFunction<
+  typeof usePermission
+>;
+
+const permission = {
+  name: 'access.something',
+  attributes: { action: 'read' as const },
+};
+
+describe('PermissionedRoute', () => {
+  it('Does not render when loading', async () => {
+    mockUsePermission.mockReturnValue({ loading: true, allowed: false });
+
+    const { queryByText } = await renderInTestApp(
+      <PermissionedRoute
+        permission={permission}
+        element={<div>content</div>}
+      />,
+    );
+
+    expect(queryByText('content')).not.toBeTruthy();
+  });
+
+  it('Renders given element if authorized', async () => {
+    mockUsePermission.mockReturnValue({ loading: false, allowed: true });
+
+    const { getByText } = await renderInTestApp(
+      <PermissionedRoute
+        permission={permission}
+        element={<div>content</div>}
+      />,
+    );
+
+    expect(getByText('content')).toBeTruthy();
+  });
+
+  it('Renders not found page if not authorized', async () => {
+    mockUsePermission.mockReturnValue({ loading: false, allowed: false });
+
+    await expect(
+      renderInTestApp(
+        <PermissionedRoute
+          permission={permission}
+          element={<div>content</div>}
+        />,
+      ),
+    ).rejects.toThrowError('Reached NotFound Page');
```

Comment:
This passes, but annoyingly shows the error message and the huge stack trace (in bright red, of course). A similar test for usePermission written down below doesn't show the error message - not sure why this one does.

Reply:
React will always print the error message unless an error boundary catches it. Not sure why `renderInTestApp` rethrows instead rendering an error page 🤔 . If you want to avoid the console error, you could wrap this in a custom error boundary and query for the error message instead (eg w/ `getByText`)

Reply:
Yeah, I'm of the notion that it should render an error page as well. I'll leave it for now, since this seems ripe for a possible wider refactor, and writing a whole new error boundary seems a bit too overkill for this.